File: backend/app/models/vehicle.py
from sqlalchemy import Column, Integer, String, Float, Date
from app.db.database import Base


class Vehicle(Base):
    __tablename__ = "vehicles"

    id = Column(Integer, primary_key=True, index=True)

    registration_number = Column(
        String,
        unique=True,
        nullable=False,
        index=True
    )

    vehicle_name = Column(
        String,
        nullable=False
    )

    vehicle_type = Column(
        String,
        nullable=False
    )

    max_load_capacity = Column(
        Float,
        nullable=False
    )

    odometer = Column(
        Float,
        default=0.0
    )

    acquisition_cost = Column(
        Float,
        nullable=False
    )

    acquisition_date = Column(
        Date
    )

    status = Column(
        String,
        default="Available"
    )

    notes = Column(
        String,
        nullable=True
    )

File: frontend/js/vehicle.js
async function loadVehicles() {

    const vehicles = await getVehicles();

    const table = document.getElementById("vehicleTable");

    table.innerHTML = "";

    vehicles.forEach(vehicle => {

        table.innerHTML += `

        <tr>

            <td>${vehicle.id}</td>

            <td>${vehicle.registration_number}</td>

            <td>${vehicle.vehicle_name}</td>

            <td>${vehicle.vehicle_type}</td>

            <td>${vehicle.status}</td>

            <td>

                <button onclick="removeVehicle(${vehicle.id})">

                    Delete

                </button>

            </td>

        </tr>

        `;

    });

}

document
.getElementById("vehicleForm")
.addEventListener("submit", async function(e){

    e.preventDefault();

    const vehicle = {

        registration_number:
            document.getElementById("registration_number").value,

        vehicle_name:
            document.getElementById("vehicle_name").value,

        vehicle_type:
            document.getElementById("vehicle_type").value,

        max_load_capacity:
            parseFloat(document.getElementById("max_load_capacity").value),

        odometer:
            parseFloat(document.getElementById("odometer").value),

        acquisition_cost:
            parseFloat(document.getElementById("acquisition_cost").value),

        acquisition_date:
            document.getElementById("acquisition_date").value,

        status: "Available",

        notes:
            document.getElementById("notes").value

    };

   try {

    await createVehicle(vehicle);

    alert("✅ Vehicle created successfully!");

    this.reset();

    loadVehicles();

} catch (error) {

    alert("❌ Failed to create vehicle.");

}

});

async function removeVehicle(id){

    await deleteVehicle(id);

alert("🗑️ Vehicle deleted successfully!");

loadVehicles();
}

loadVehicles();